packages/itwin/ec3-widget/src/test/DeleteModal.test.tsx

```diff
@@ -0,0 +1,106 @@
+/*---------------------------------------------------------------------------------------------
+* Copyright (c) Bentley Systems, Incorporated. All rights reserved.
+* See LICENSE.md in the project root for license terms and full copyright notice.
+*--------------------------------------------------------------------------------------------*/
+import React from "react";
+import "@testing-library/jest-dom";
+import { screen } from "@testing-library/react";
+import { renderWithContext, simulateClick } from "./test-utils";
+import { DeleteModal } from "../components/DeleteModal";
+
+describe("DeleteModal", () => {
+  it("Delete modal with the show prop should be visible", async () => {
+    await renderWithContext({
+      component: <DeleteModal
+        entityName="template"
+        show={true}
+        setShow={() => { }}
+        onDelete={async () => { }}
+        refresh={async () => { }}
+      />,
+    });
+    expect(screen.getByTestId("ec3-delete-modal")).toBeDefined();
+    expect(document.querySelectorAll(".iui-dialog-visible")).toBeDefined();
+  });
+
+  it("Delete modal without the show prop should not be visible", async () => {
+    await renderWithContext({
+      component: <DeleteModal
+        entityName="template"
+        show={false}
+        setShow={() => { }}
+        onDelete={async () => { }}
+        refresh={async () => { }}
+      />,
+    });
+    expect(document.querySelector(".ec3-delete-modal")).toBeDefined();
+    expect(document.querySelector(".iui-dialog-visible")).toBeNull();
+  });
+
+  it("Closing using button in top right calls setShow function", async () => {
+    const show = jest.fn();
+    await renderWithContext({
+      component: <DeleteModal
+        entityName="template"
+        show={true}
+        setShow={show}
+        onDelete={async () => { }}
+        refresh={async () => { }}
+      />,
+    });
+    expect(screen.getByTestId("ec3-delete-modal")).toBeDefined();
+    expect(document.querySelectorAll(".iui-dialog-visible")).toBeDefined();
+
+    const button = document
+      .querySelector(".iui-dialog-title-bar")
+      ?.querySelector(".iui-button") as HTMLInputElement;
```

Comment:
Isn't there anything a bit more specific for this button?

Reply:
This button is deep inside the ITwin React Component so I can't give it a test-id. There is only one title bar in the modal and the title bar only has one button so the current selection does narrow it down to the single button.